brave/brave-core
# Synchronize brave vpn system tray with the browser and the wireguard service

Resolves https://github.com/brave/brave-browser/issues/30708

- Added main menu item to show/hide tray icon for brave vpn
- Added menu to the tray icon to manage wireguard vpn connection
- Updated the service watcher to be able to subscribe to different events by mask
- Auto start tray icon process when wireguard vpn connected
   - The tray process manages cases to show/hide icon inside itself
   - It checks if already started or user made icon hidden or no last used config yet.

https://github.com/brave/brave-core/assets/2965009/3066386c-9c22-45a8-8b4b-c946ca72a5a3




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

- Enable wireguard vpn brave://flags/#brave-vpn-wireguard
- Connect to BraveVpn in the browser and check the tray place for the BraveVpn icon.
- Play with the tray icon menu and with the browser vpn connection toggler, try to connect/disconnect. The tray icon should show all actual statuses for the connection.
- Check for different windows themes, the dark theme should have own icons.
- If user clicked to remove the tray icon then it can be enabled back from the browser main submenu for BraveVpn.
